event-store/src/lib.rs

```diff
@@ -93,32 +93,36 @@ where
     }
 
     /// Query the backing store and return an entity `T`, reduced from queried events
-    fn aggregate<'b, E, T, A>(&self, query_args: A) -> BoxedFuture<'b, T, String>
+    fn aggregate<'b, E, T, A>(&'b self, query_args: A) -> BoxedFuture<'b, T, String>
```

Comment:
Can you see if you can get rid of this lifetime? I feel like putting explicit lifetimes on `self` is a big code smell. I could be wrong and it's required, but it'd be killa if it wasn't.

Reply:
Addendum: there's [this SO question](https://stackoverflow.com/questions/31641610/return-reference-with-lifetime-of-self) which seems to suggest `'b` can be elided. Would that work?